pkg/plugins/manager/loader/finder/local.go

```diff
@@ -79,14 +80,23 @@ func (l *Local) Find(ctx context.Context, src plugins.PluginSource) ([]*plugins.
 			continue
 		}
 
+		// Skip core plugins if the feature flag is enabled and the plugin is in the skip list.
+		// It could be loaded later as an external plugin.
+		if l.cfg.Features != nil && l.cfg.Features.IsEnabled(featuremgmt.FlagDecoupleCorePlugins) &&
```

Comment:
this is ideally a filter (in the pipeline scenario, as you mentioned). My idea was to avoid any possible processing if the plugin is meant to be skipped but in any case I can move it in the meantime.